### DISCUSSION TITLE
[@types/node] module.constructor._extensions is not defined

### DISCUSSION THREAD

**Discussion by SamuelCharpentier** [DefinitelyTyped/DefinitelyTyped#61561] [Issues with a @types package]
I'm getting the followin error
`error TS2339: Property '_extensions' does not exist on type 'Function'.`

While trying to compile the following ts file:
```typescript
const noExports = Symbol('no exports');
const extensions = module.constructor._extensions; //Property '_extensions' does not exist on type 'Function'.
const jsLoader = extensions['.js'];
extensions['.js'] = (module: any, filename: any) => {
	module.exports[noExports] = filename;
	jsLoader(module, filename);
};

const help1 = require('./helpers-without-exports.js');
const help2 = require('./helpers-with-empty-exports.js');

console.log(noExports in help1); // true
console.log(noExports in help2); // false
```
My only dependencies are
```json
"@types/node": "^18.6.3",
 "typescript": "^4.7.4"
```
Using node v16.16.0 and the basic tsconfig.json with `npx tsc --init`. Is there additional tweaking needed to get this to compile?  The equivalent in vanilla JS runs as expected.

[a comment by an automated account is not shown]

**Comment by SimonSchick** [comment]
I don't think that's public API (and we probably wont add it to type defs), are you trying to use https://nodejs.org/api/modules.html#requireextensions instead?

**Reply by SamuelCharpentier** [reply]
I mean, it's `Deprecated since: v0.10.6`. How could it help me find out if a module is exported as an empty object or has no module exported in it?
